platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/IvyArtifact.java

```diff
@@ -16,81 +16,49 @@
 
 package org.gradle.api.publish.ivy;
 
+import org.gradle.api.provider.Property;
 import org.gradle.api.publish.PublicationArtifact;
-import org.gradle.internal.instrumentation.api.annotations.ToBeReplacedByLazyProperty;
-
-import javax.annotation.Nullable;
+import org.gradle.api.tasks.Optional;
+import org.gradle.internal.instrumentation.api.annotations.ReplacesEagerProperty;
 
 /**
  * An artifact published as part of a {@link IvyPublication}.
  */
 public interface IvyArtifact extends PublicationArtifact {
     /**
-     * The name used to publish the artifact file, never <code>null</code>.
+     * The name used to publish the artifact file.
      * Defaults to the name of the module that this artifact belongs to.
      */
-    @ToBeReplacedByLazyProperty
-    String getName();
-
-    /**
-     * Sets the name used to publish the artifact file.
-     * @param name The name.
-     */
-    void setName(String name);
+    @ReplacesEagerProperty
+    Property<String> getName();
 
     /**
-     * The type used to publish the artifact file, never <code>null</code>.
+     * The type used to publish the artifact file.
      */
-    @ToBeReplacedByLazyProperty
-    String getType();
-
-    /**
-     * Sets the type used to publish the artifact file.
-     * @param type The type.
-     */
-    void setType(String type);
+    @ReplacesEagerProperty
+    Property<String> getType();
 
     /**
      * The extension used to publish the artifact file, never <code>null</code>.
      * For an artifact without an extension, this value will be an empty String.
      */
-    @ToBeReplacedByLazyProperty
-    String getExtension();
-
-    /**
-     * Sets the extension used to publish the artifact file.
-     * @param extension The extension.
-     */
-    void setExtension(String extension);
+    @ReplacesEagerProperty
+    Property<String> getExtension();
 
     /**
      * The classifier used to publish the artifact file.
-     * A <code>null</code> value (the default) indicates that this artifact will be published without a classifier.
-     */
-    @Nullable
-    @ToBeReplacedByLazyProperty
-    String getClassifier();
-
-    /**
-     * Sets the classifier used to publish the artifact file.
-     * @param classifier The classifier.
+     * An absent value (the default) indicates that this artifact will be published without a classifier.
      */
-    void setClassifier(@Nullable String classifier);
+    @Optional
```

Comment:
💅 It looks this is not an input so `@Optional` won't do anything here

Reply:
I think it would be useful for anyone who reads the javadoc. Another option would be to replace it with explicit text that classifier might be null